myserver.cpp

```diff
@@ -0,0 +1,684 @@
+#include "myserver.h"
+#include "IDataBase.cpp"
+#include "IDataBase.h"
+#include "DBErrors.h"
+#include <iostream>
+#include <vector>
+#include <string>
+
+myserver::myserver(){}
+
+myserver::~myserver(){}
+
+void myserver::startServer(){
+    if(this->listen(QHostAddress::Any,5555)){
+        qDebug()<<"Listening";
+    }
+    else{
+        qDebug()<<"Not listening";
+    }
+}
+
+void myserver::incomingConnection(qintptr pisocketDescriptor){
+    socket = new QTcpSocket(this);
+    socket->setSocketDescriptor(pisocketDescriptor);
+
+    connect(socket, SIGNAL(readyRead()),this,SLOT(sockReady()));
+    connect(socket, SIGNAL(disconnected()),this,SLOT(sockDisc()));
+
+    qDebug()<<pisocketDescriptor<<"Client connected";
+
+    socket->write("{\"type\":\"connect\",\"status\":\"yes\"}");
+    qDebug()<<"Send client connect status - YES";
+}
+
+void myserver::sockReady(){
+    Data = socket->readAll();
+    qDebug()<<"Data: "<<Data;
+    std::vector <std::string> msg;
+
+    nlohmann::json j;
+    j = nlohmann::json::parse(QString(Data).toStdString());
+
+    for (nlohmann::json::iterator it = j.begin(); it != j.end(); ++it) {
+            msg.push_back(it.value());
+    }
+    QJsonDocument document;
+    QByteArray bytes;
+
+        if(msg[0]=="AddUser"){
```

Comment:
Try to use some function with check if this elements msg[0] exists like at()
Or add your own check.

Reply:
Fixed it. Added at()